### ISSUE TITLE
Classification standardization and potential flexibility with number of predictive classes

### ISSUE THREAD

**Issue by mayhemsloth** [mayhemsloth/Drum-Tabber#6]
Once a training set has been created, or at least the total labels have been loaded fully, then a standardization needs to be applied to the entire training set. Similar to how the drum piece labels needed a standardized labeling system (the master format) to properly compare drum tabs across multiple songs, the class labels also need a proper master format. 

**The total amount of distinct class labels will ultimately determine how many output classes there will be for the predictive model**. For example, in most tabs there are usually two "volumes" denoted for different drum pieces: a lower case letter (o on drums or x on cymbals) for normal, softer hits, and the upper case letter (O on drums or X on cymbals) for harder, louder hits. Often though, for cymbals especially, there isn't much distinction between the two hits and the usage consistency across multiple tabs is simply not present. Collapsing these two classifications into one classification would require some processing of the entire label data set. 

By analogy, lots of other classifications would need to be collapsed due to labeling inconsistency and/or simplification of the model itself. For example, the hi-hat can usually be hit closed (x), slightly open - also called washy (X or w or W), or completely open (o or O, on the hi-hat). Almost certainly my limited training data will not be sufficient to train the model on *multiple classes* of a *single* drum piece. For the mvp, I will very likely collapse all hits on each drum piece label down to *one* class, and also most likely combine all the crash cymbals (CC, C2, C3, C4, SP, CH) into one single class as well because cymbals are extremely hard to distinguish by human ear *and* the consistency is completely absent. 

So to complete this issue, I need to have code written and working that converts the entire label set to my liking, at least for the mvp. Further work can be done to expand the classes if necessary (that is, specifically choose a wider scope of classes).

**Comment by mayhemsloth** [comment]
The `clean_labels` and `collapse_class` functions have been implemented in the "Batch Processing and Classification Standardization" Jupyter notebook. 

The`clean_labels` function doesn't have any user choices, because that is mainly where automatic cleaning of labels occurs. Choices are technically hard coded in and can be changed inside the function, but most are reasonable for the current scope of the project, even if a 10-100x data set is produced. 

The `collapse_class` function has a bunch of options based on arguments to the function and all are mutually independent (that is, all possible subsets of argument choices should work). Defaults are set to limit classes, but the function can return more classes if necessary. With my initial training set I will be collapsing the classes down to bass drum, snare drum, hihat (closed), all cymbals, and all toms: 5 drum classes and then 2 beat classes derived from the time-keeping line (down-beat and off-beat).
